runtime/src/staking.rs

```diff
@@ -1085,6 +1103,76 @@ decl_module! {
             Self::deposit_event(RawEvent::PermissionedValidatorStatusChanged(controller));
         }
 
+        /// Validate the nominators KYC expiry time
+        ///
+        /// If an account from a given set of address is nominating then
+        /// check the KYC expiry time of it and if it is expired
+        /// then the account should be unbonded and removed from the nominating process.
+        ///
+        /// #<weight>
+        /// - Depends on passed list of AccountId
+        /// - Depends on the no. of claim issuers an accountId has for the KYC expiry
+        /// #</weight>
+        #[weight = SimpleDispatchInfo::FixedNormal(950_000)]
+        fn validate_kyc_expiry_nominators(origin, targets: Vec<T::AccountId>) {
+            let caller = ensure_signed(origin)?;
+            let mut expired_nominators = Vec::new();
+            ensure!(!targets.is_empty(), "targets cannot be empty");
+            // Iterate provided list of accountIds (These accountIds should be stash type account)
+            for i in 0..targets.len() {
+                // Check whether given nominator is vouching for someone or not
+
+                if !(Self::nominators(&targets[i])).is_empty() {
+                    // Access the identity of the nominator
+                    if let Some(nominate_identity) = <identity::Module<T>>::get_identity(&(Key::try_from(targets[i].encode())?)) {
+                        // Access all the claim issuers for a given nominator
+                        // then iterate all of them which provide the claim for KYC expiration.
+                        // There is a possibility that nominator will have more than one claim for the same key,
+                        // So we iterate all of them and if any one of the claim value doesn't expire then nominator posses
+                        // valid KYC otherwise it will be removed from the pool of the nominators.
+                        let claim_issuers = <identity::Module<T>>::claim_issuers(nominate_identity);
+                        let mut count: u32 = 0;
```

Comment:
We will have a fixed list of KYC providers (claim issuers) who are trusted for this claim (KYC_EXPIRY_CLAIM_KEY) rather than trusting any issuer.

The list of claim issuers that are trusted is added in https://github.com/PolymathNetwork/Polymesh/pull/148

I think this function (that checks that an identity has a valid KYC claim from a trusted KYC SP) should be somewhere like the identity module as we will use it in other places as well.